### ISSUE TITLE
Lack of UEFI support should be Documented

### ISSUE THREAD

**Issue by Frimbleglim** [subgraph/subgraph-os-issues#192]
Hi, I have noticed that the ISO does not support EFI boot.  

This should probably be in your documentation.  It's fairly key to whether the OS will work on a given computer.

**Comment by ghost** [comment]
Yeah this is pretty important for much modern hardware as even Debian and Ubuntu support EFI now.

**Comment by burdges** [comment]
I'd suggest adding a UEFI kernel online someplace that comes with big scary warnings about whatever risks it creates.

**Comment by xSmurf** [comment]
@burdges FWIW lack of UEFI isn't from the kernel, but troubles with the Debian installer.

@bkerensa AFAIK, all modern hardware still supports booting in "legacy" mode.

**Comment by conorsch** [comment]
Was bitten by this as well, and have already had to pass on the advice to set legacy boot mode via oral tradition. Do consider documenting the hard dependency on the [Subgraph OS pre-Alpha for Testers](https://support.subgraph.com/private/sgos/pre-alpha/) page.

**Comment by CrustyDude** [comment]
If all of those things are involved, how do I install sgos on my already running Win 8.1 pc in a way that it actually allows me to boot sgos?

Thank you in advance

**Comment by mckinney-subgraph** [comment]
As of the next ISO, we should have UEFI support on par with (but not better than) Debian Stretch or other derivatives such as Kali Linux. I will update and close this issue when we release it.

**Comment by epidemics-scepticism** [comment]
_Documenting this to help anyone with issues as a temporary workaround._

I have an EFI-only only bootloader (modified chromebook), so legacy boot wasn't an option for me. I installed EFI with a few manual steps:

1. During the install process, manually partitioned the disk. 512MB for /boot, 512MB for ESP (EFI System Partition), rest for whatever you need.
2. Go through the install process normally, for me it reported that installing grub failed, skip past it and continue the installation.
3. After the installation is complete, rebooted into the subgraph live mode.
4. Pop a terminal, and mount your `/` filesystem over `/mnt` and your boot filesystem over `/mnt/boot` and the ESP filesystem over `/mnt/boot/efi`.
5. Then bind mount other files (/sys/ is important for efivars): `for i in /dev /dev/pts /proc /sys /run; do sudo mount -B $i /mnt$i; done`
6. `chroot /mnt`
7. `apt update && apt install grub-efi-amd64-bin efibootmgr` to get the required packages (**N.B.** don't try to install `grub-efi-amd64`, it will cause conflicts and break stuff, just get the `-bin` package)
7. `grub-install --target=x86_64-efi --efi-directory=/boot/efi --bootloader-id=subgraph /dev/sdx`, replacing /dev/sdx with the disk you are installing subgraph too.
8. `update-grub`
9. Leave the chroot (ctrl-d) and `umount -R /mnt` then reboot.

Gave me a working EFI installation of Subgraph.

**Comment by burdges** [comment]
Subgraph on Chromebooks, and other low end machines, is looking increasingly attractive with the restrictions on traveling with laptops.
